File: 11059.py
def isCree(s):
    dis = len(s) // 2

    a = 0
    for i in range(dis):
        a += int(s[i])

    b = 0
    for i in range(dis, len(s)):
        b += int(s[i])

    if a == b: return True
    else: return False

s = input()

l = len(s)
if l % 2: l -= 1
while 1:
    found = False
    for i in range(len(s) - l + 1):
        if isCree(s[i:i+l]):
            print(l)
            found = True
            break

    if found: break

    l -= 2

File: 11059.cpp
#include <iostream>
using namespace std;

bool isCree(string s) {
	int a = 0, b = 0;
	for (int i = 0; i < s.length() / 2; i++)
		a += s[i] - '0';
	for (int i = s.length() / 2; i < s.length(); i++)
		b += s[i] - '0';

	if (a == b) return true;
	else return false;
}

int main() {
	string s; cin >> s;

	int len = s.length();
	if (len % 2) len--;
	while (1) {
		bool found = false;
		for (int i = 0; i <= s.length() - len; i++) {
			if (isCree(s.substr(i, len))) {
				cout << len << endl;
				found = true;
				break;
			}
		}

		if (found) break;

		len -= 2;
	}
}
